extensions/replication/queueProcessor/MultipleBackendTask.js

```diff
@@ -0,0 +1,175 @@
+const async = require('async');
+
+const errors = require('arsenal').errors;
+const jsutil = require('arsenal').jsutil;
+
+const QueueProcessorTask = require('./QueueProcessorTask');
+const attachReqUids = require('../utils/attachReqUids');
+
+class MultipleBackendTask extends QueueProcessorTask {
+
+    _setupRolesOnce(entry, log, cb) {
+        log.debug('getting bucket replication', { entry: entry.getLogInfo() });
+        const entryRolesString = entry.getReplicationRoles();
+        let entryRoles;
+        if (entryRolesString !== undefined) {
+            entryRoles = entryRolesString.split(',');
+        }
+        if (entryRoles === undefined || entryRoles.length !== 1) {
+            log.error('expecting a single role in entry replication ' +
+            'configuration when replicating to an external location', {
+                method: 'MultipleBackendTask._setupRolesOnce',
+                entry: entry.getLogInfo(),
+                roles: entryRolesString,
+            });
+            return cb(errors.BadRole);
+        }
+        this.sourceRole = entryRoles[0];
+
+        this._setupSourceClients(this.sourceRole, log);
```

Comment:
I would recommend calling this explicitly, as this method is doing more than just `_setupRolesOnce`.

Reply:
Do you mean moving this out of _setupRolesOnce?

Reply:
Yes.